crates/kitsune_p2p/Makefile

```diff
@@ -0,0 +1,50 @@
+STATIC_CHECKS := static-fmt static-toml static-clippy static-doc
+
+.PHONY: $(STATIC_CHECKS) format build build-each test
+
+# Run all static checks
+check: $(STATIC_CHECKS)
+
+# Check formatting
+static-fmt:
+	cargo fmt --check
+
+# Check TOML formatting
+static-toml:
+	taplo format --check ./*.toml
+	taplo format --check ./crates/**/*.toml
+
+# Ensure our chosen style lints are followed
+static-clippy:
+	CHK_SQL_FMT=1 cargo clippy --all-targets -- --deny warnings
+
+# Check that the docs for this workspace can by generated without warnings
+static-doc:
+	RUSTDOCFLAGS="-D warnings" cargo doc --no-deps
+
+# Format Rust and TOML files
+format:
+	cargo fmt
+	taplo format ./*.toml
+	taplo format ./crates/**/*.toml
+
+# Build the workspace
+build:
+	cargo build \
+		--workspace \
+		--locked \
+		--all-targets
+
+# Build each crate in the workspace, to find issues with merged features
+PROJECTS=$(wildcard crates/*)
+build-each:
+	@$(foreach PROJ,$(PROJECTS),\
+		cd $(PROJ); cargo build --locked; cd ../.. ; \
+		)
```

Comment:
I'm worried this is going to cause build times and cache sizes to both be large, but we can see and adjust as needed : )

Reply:
At least for build times, because this is running in parallel with the Holochain builds, I don't think that's an issue for now. If grows the cache unreasonably then yes, we might need to adjust.